rivershared/sqlctemplate/sqlc_template.go

```diff
@@ -110,9 +110,19 @@ func (r *Replacer) Run(ctx context.Context, sql string, args []any) (string, []a
 // RunSafely is the same as Run, but returns an error in case of missing or
 // extra templates.
 func (r *Replacer) RunSafely(ctx context.Context, sql string, args []any) (string, []any, error) {
-	// If nothing present in context, short circuit quickly.
-	container, containerOK := ctx.Value(contextKey{}).(*contextContainer)
-	if !containerOK {
+	var (
+		container, containerOK = ctx.Value(contextKey{}).(*contextContainer)
+		sqlContainsTemplate    = strings.Contains(sql, "/* TEMPLATE")
+	)
+	switch {
+	case containerOK && !sqlContainsTemplate:
+		return "", nil, errors.New("sqlctemplate found context container but SQL contains no templates; bug?")
+
+	case !containerOK && sqlContainsTemplate:
+		return "", nil, errors.New("sqlctemplate found template(s) in SQL, but no context container; bug?")
+
+	case !containerOK && !sqlContainsTemplate:
+		// Neither context container or template in SQL; short circuit fast because there's no work to do.
 		return sql, args, nil
```

Comment:
Switch statements are executed top-to-bottom, so maybe you want this fast path short circuit to go first?

Reply:
Done. Thx.